[chore] Add `README` to the project (#15)
nimblehq/github-action-runner-android: README.md

## Setup self-hosted runner

### Prerequisites

- Machine running macOS from [MacStadium] or a local one.
- Android Studio installation is recommended to facilitate the installation of other required programs such as Java, Android SDK, and AVD (needed for running the UI test).

[MacStadium]: https://www.macstadium.com/

### Add a self-hosted runner to GitHub repository

1. Open the **Settings** menu of the repository. Go to **Actions** -> **Runners** -> click on the **Add runner** button.
2. Make sure the Operating System dropdown shows **macOS** as a selected OS, then run all of the listed commands from the instruction in your machine respectively and specify configurations when prompted.
3. (Optional) In the directory where the runner was installed, run the following command to install **svc**, which is a program that can help to check and manage the status of the runner.

   ```
   ./svc.sh install
   ```

4. In the directory where the runner was installed, run the following command to launch and inform the runner to listening for the upcoming jobs.

   ```
   ./run.sh
   ```

### Giving the ability to run on a self-hosted runner to workflows

> Define the following key to all the jobs that we target to run on a self-hosted runner.

Author: ```suggestion
Define the following key to all the jobs that we target to run on a self-hosted runner:
```

# GitHub Actions Runner for Android

The repository for proof of concept of [Custom Runner for Github Actions (Android) initiative](https://www.notion.so/nimblehq/Custom-Runner-for-Github-Actions-Android-bb4c0b6310184e44afe9b3bea0989a57?d=72eb950e-33be-482b-8e17-40c92023073d)

## Setup custom runner

### Prerequisites

- Machine running macOS from [MacStadium](https://www.macstadium.com/) or a physical one

> - Android Studio installation is recommended as it will help to install all the required Java, Android
>   SDK, also the AVD which needed for running UI test

Author: ```suggestion
- Android Studio installation is recommended to facilitate to install all the required Java, Android SDK, also the AVD which needed for running the UI test
```

> ### Add custom (self-hosted) runner

Author: ```suggestion
### Add custom (self-hosted) runner to Github repository
```

I think it might create confusion to reader if we don't mention to Github repository here.

1. Open the **Settings** menu of the repository. Go to **Actions** -> **Runners** -> click on the **Add runner** button
2. Make sure the Operating System dropdown shows **macOS** as a selected item, then run all of the below commands in your machine in order. Specify configurations when prompted.

> 3. (Optional) In the directory where the runner was installed, run `./svc.sh install`. This program is for help checking the status of the runner.

Author: I think this format is more readable for the document:

3. (Optional) In the directory where the runner was installed, run 

   ```shell
   ./svc.sh install
   ``` 

   This program is for help checking the status of the runner.

   ------------------------------------------------------------------------
I cannot make the suggestion block here because there's ``` sign 😂 
it is  
![Screen Shot 2564-08-19 at 15 31 22](https://user-images.githubusercontent.com/12026942/130035935-84593e4c-2b1d-4ccf-a7c1-7e3427b93dc8.png)

Author: I'm not sure if `below commands` phrase is understandable enough. 🤔  Shall we add something like `below commands from Github instruction`?

> 4. In the directory where the runner was installed, run `./run.sh` to open the runner to listening for the upcoming jobs.

Author: Same here.

4. In the directory where the runner was installed, run 

   ```shell
   ./run.sh
   ```
    
   to open the runner to listening for the upcoming jobs.

### Giving the ability to run on a self-hosted runner to workflow

> Define `runs-on: self-hosted` to all the jobs we want to be run on a self-hosted runner.

Author: ```suggestion
Define this following key to all the jobs we target to run on a self-hosted runner.

```yml
runs-on: self-hosted
//```
```

### Known issues

> - One runner can only execute one job at a time. To run multiple jobs or workflows simultaneously, we might need to consider adding a new runner in the same or different machine.
> - There is an issue related to `actions/cache` in a low-performance machine as the default `upload-chunk-size` value is too big for such a machine. Consider reducing this value when facing such an issue.
> - If there's an error related to being incapable of finding the Android SDK. We might need to specify the Android SDK path into `ANDROID_SDK_ROOT` environment variable, for example: `export ANDROID_SDK_ROOT=/Users/username/Library/Android/sdk`.
> - In a low-performance machine, Creating an AVD from scratch every time running the Android emulator might be a better solution in terms of execution time as the speed for retrieving cache is low in such a machine.

Author: ```suggestion
- One runner can only execute one job at a time. To run multiple jobs or workflows simultaneously, we need to consider adding a new runner in the same or different machine.
- There is an issue related to `actions/cache` in a low-performance machine since the default `upload-chunk-size` value is too big. So, we can consider reducing this value when facing the slow computing issue.
- In case there's an error related to being incapable of finding the Android SDK. We can resolve it by specifying the Android SDK path into `ANDROID_SDK_ROOT` environment variable, for example: `export ANDROID_SDK_ROOT=/Users/username/Library/Android/sdk`.
- In a low-performance machine, Creating an AVD from scratch every time running the Android emulator might be a better solution in terms of execution time as the speed for retrieving cache is low in such a machine.
```

Author: Should we add this as the 2nd bullet?
```
- In case we have multiple runner machines, it will run all jobs or workflows parallelly by dispersing the workload to all of the machine. If we want to fix the machine to run especially job, we can add the label into the machine and specify to run on that label instead of self-hosted
```

<p align="center">
  <img alt="Nimble logo" src="https://assets.nimblehq.co/logo/light/logo-light-text-320.png" />

> </p>

Reviewer: @lydiasama @chornerman I am not sure about removing the header logo and steps to set up a new project from the template 🤔

Reviewer 2: @hoangnguyen92dn I Bring back all the parts from the Android template and stuck my part under it. Please let me know if you have more suggestions.

Reviewer: @chornerman 🤔  This repo is for Github Action runner so I think you could put it on top of the step-up steps. Both are good to me 😄
